core/src/style/pointStyle.cpp

```diff
@@ -52,36 +54,51 @@ void PointStyle::constructShaderProgram() {
     }
 
     m_shaderProgram->addSourceBlock("defines", defines);
+
+    m_mesh = std::make_unique<LabelMesh>(m_vertexLayout, m_drawMode);
+}
+
+void PointStyle::onBeginFrame() {
+    // Upload meshes for next frame
+    m_mesh->upload();
 }
 
 void PointStyle::onBeginDrawFrame(const View& _view, Scene& _scene) {
     Style::onBeginDrawFrame(_view, _scene);
 
+    auto texUnit = RenderState::nextAvailableTextureUnit();
+
     if (m_spriteAtlas) {
-        m_spriteAtlas->bind(RenderState::nextAvailableTextureUnit());
+        m_spriteAtlas->bind(texUnit);
     } else if (m_texture) {
-        m_texture->update(RenderState::nextAvailableTextureUnit());
-        m_texture->bind(RenderState::currentTextureUnit());
+        m_texture->update(texUnit);
+        m_texture->bind(texUnit);
     }
 
-    m_shaderProgram->setUniformi(m_uTex, 0);
+    m_shaderProgram->setUniformi(m_uTex, texUnit);
     m_shaderProgram->setUniformMatrix4f(m_uOrtho, _view.getOrthoViewportMatrix());
 
+    m_mesh->draw(*m_shaderProgram);
+    m_mesh->clear();
 }
 
 struct PointStyleBuilder : public StyleBuilder {
 
+    struct PointStyleMesh : public LabelMesh, public LabelSet {
+        using LabelMesh::LabelMesh;
+    };
```

Comment:
Ah, yeah, LabelMesh has changed in between. And after unifying text- and sprite rendering it can extends StyledMesh instead of LabelMesh. Updating it now